hack/generated/controllers/suite_test.go

```diff
@@ -41,19 +55,101 @@ func (tc *ControllerTestContext) SharedResourceGroupOwner() genruntime.KnownReso
 	return genruntime.KnownResourceReference{Name: tc.SharedResourceGroup.Name}
 }
 
-func setup() error {
+func setupEnvTest() (*rest.Config, error) {
+	envtestContext = &EnvtestContext{
+		testenv: envtest.Environment{
+			CRDDirectoryPaths: []string{
+				"../config/crd/bases/valid",
+			},
+		},
+		stopManager: make(chan struct{}),
+	}
+
+	log.Print("Starting envtest")
+	config, err := envtestContext.testenv.Start()
+	if err != nil {
+		return nil, errors.Wrapf(err, "starting envtest environment")
+	}
+
+	log.Print("Creating & starting controller-runtime manager")
+	mgr, err := ctrl.NewManager(config, ctrl.Options{Scheme: testcommon.CreateScheme()})
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating controller-runtime manager")
+	}
+
+	envtestContext.manager = mgr
+
+	go func() {
+		// this blocks until the input chan is closed
+		err := envtestContext.manager.Start(envtestContext.stopManager)
+		if err != nil {
+			log.Fatal(errors.Wrapf(err, "running controller-runtime manager"))
+		}
+	}()
+
+	log.Print("Creating ARM client")
+	armClient, err := armclient.NewAzureTemplateClient()
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating ARM client")
+	}
+
+	log.Print("Registering custom controllers")
+	errs := controllers.RegisterAll(
+		envtestContext.manager,
+		armClient,
+		controllers.KnownTypes,
+		klogr.New(),
+		controller.Options{})
```

Comment:
@matthchr should we be setting anything here?

Reply:
Looks like standard usage is showing just `concurrency(1)`, so I'm thinking no?